extensions/elasticsearch-rest-client-common/deployment/src/main/java/io/quarkus/elasticsearch/restclient/common/deployment/DevServicesElasticsearchProcessor.java

```diff
@@ -189,17 +187,25 @@ private DevServicesResultBuildItem.RunningDevService startElasticsearch(
             }
         }
 
+        if (buildItemConfig.distribution != null && !buildItemConfig.distribution.equals(config.distribution)) {
+            throw new BuildException(
+                    "Dev Services for Elasticsearch detected a distribution mismatch, distribution is " + config.distribution
+                            + " but the configured distribution is " + buildItemConfig.distribution +
+                            ". Either configure a different distribution or disable Dev Services for Elasticsearch.",
```

Comment:
Maybe this would be a bit better:
```suggestion
                            ". Either configure the same distribution or disable Dev Services for Elasticsearch.",
```